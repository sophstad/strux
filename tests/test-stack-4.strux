int main() {
  Stack::int:: si = new Stack::int::();
  si.add(0);
  si.add(1);
  si.add(2);
  si.add(3);
  si.add(4);
  si.show();

  Stack::num:: sn = new Stack::num::();
  sn.add(326.3);
  sn.add(-1752552844.3);
  sn.add(98.0);
  sn.show();

  Stack::string:: ss = new Stack::string::();
  ss.add("Sophie");
  ss.add("Strux");
  ss.add("Kofi");
  ss.add("Millie");
  ss.add("Josh");
  ss.show();

  Stack::bool:: sb = new Stack::bool::();
  sb.add(true);
  sb.add(false);
  sb.add(true);
  sb.show();

  return 0;
}
