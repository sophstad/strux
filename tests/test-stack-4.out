+-------------+ <- Top
| 4           |
+-------------+
| 3           |
+-------------+
| 2           |
+-------------+
| 1           |
+-------------+
| 0           |
+-------------+
+--------------------+ <- Top
| 98.000000          |
+--------------------+
| -1752552844.300000 |
+--------------------+
| 326.300000         |
+--------------------+
+------------------------------------------------+ <- Top
| Josh                                           |
+------------------------------------------------+
| Millie                                         |
+------------------------------------------------+
| Kofi                                           |
+------------------------------------------------+
| Strux                                          |
+------------------------------------------------+
| Sophie                                         |
+------------------------------------------------+
+-------------+ <- Top
| 1           |
+-------------+
| 0           |
+-------------+
| 1           |
+-------------+
